src/Common/Caching/CacheClient.cs

```diff
@@ -419,6 +466,14 @@ protected void CreateParentDirectory(string filePath)
         }
 
         return (pathSet, nodeBuildResult);
+
+        static bool AreOnSameDrive(string path1, string path2)
```

Comment:
@erikmav what is the canonical way to do this?

Reply:
Complicated by `subst` as well as drive mounts and other obfuscations, see the complicated logic in the CoW library at https://github.com/microsoft/CopyOnWrite/blob/main/lib/Windows/VolumeEnumerator.cs . I would cut this method and the call to it and instead use `_copyOnWriteFilesystem.CopyOnWriteLinkSupportedBetweenPaths(sourceAbsolutePath, destinationAbsolutePath, CloneFlags.PathIsFullyResolved)` where it is called above. The logic in the CoW library has cached info and a fast lookup for each path to make it approximately as fast as checking drive letters in the 99% case.